encrypted-media/scripts/playback-temporary-expired.js

```diff
@@ -0,0 +1,84 @@
+function runTest(config,qualifier) {
+
+    var testname = testnamePrefix(qualifier, config.keysystem)
+                                    + ', temporary, '
+                                    + /video\/([^;]*)/.exec(config.videoType)[1]
+                                    + ', expired license';
+
+    var configuration = {   initDataTypes: [config.initDataType],
+                            audioCapabilities: [{contentType: config.audioType}],
+                            videoCapabilities: [{ contentType: config.videoType}],
+                            sessionTypes: ['temporary'] };
+
+    async_test(function(test) {
+
+        var _video = config.video,
+            _mediaKeys,
+            _mediaKeySession,
+            _mediaSource;
+
+        function onFailure(error) {
+            forceTestFailureFromPromise(test, error);
+        }
+
+        function onEncrypted(event) {
+            assert_equals(event.target, _video);
+            assert_true(event instanceof window.MediaEncryptedEvent);
+            assert_equals(event.type, 'encrypted');
+
+            // Only create the session for the firs encrypted event
+            if (_mediaKeySession !== undefined) return;
+
+            var initDataType = config.initData ? config.initDataType : event.initDataType;
+            var initData = config.initData || event.initData;
+
+            _mediaKeySession = _mediaKeys.createSession('temporary');
+            waitForEventAndRunStep('message', _mediaKeySession, onMessage, test);
+            _mediaKeySession.generateRequest( initDataType, initData ).catch(onFailure);
+        }
+
+        function onMessage(event) {
+            assert_equals(event.target, _mediaKeySession);
+            assert_true(event instanceof window.MediaKeyMessageEvent);
+            assert_equals(event.type, 'message');
+
+            assert_in_array(event.messageType, ['license-request', 'individualization-request']);
+
+            var expiration = Date.now().valueOf();
+            config.messagehandler(event.messageType, event.message, expiration).then(function(response) {
+                return event.target.update(response);
+            }).then(test.step_func(function(){
+                assert_approx_equals(event.target.expiration, expiration, 2000, "expiration attribute should equal provided expiration time");
+                setTimeout(test.step_func(function() {
```

Comment:
It is not recommended to use [setTimeout](https://github.com/w3c/testharness.js/blob/master/docs/api.md#timeouts-in-tests). Could you use step_timeout instead of setTimeout?